frontend/javascripts/components/select_experience_domain.js

```diff
@@ -45,21 +48,45 @@ class SelectExperienceDomain extends React.PureComponent<Props, State> {
     return this.state.domains.filter(domain => !this.props.alreadyUsedDomains.includes(domain));
   }
 
+  onSearch = (domain: string) => {
+    this.setState({ currentlyEnteredDomain: domain });
+  };
+
   render() {
+    const {
+      value,
+      notFoundContent,
+      width,
+      disabled,
+      placeholder,
+      onSelect,
+      onChange,
+      allowCreation,
+    } = this.props;
+    const { currentlyEnteredDomain } = this.state;
+    const options = this.getUnusedDomains();
+    if (
+      allowCreation &&
+      !options.includes(currentlyEnteredDomain) &&
+      currentlyEnteredDomain.trim() !== ""
+    ) {
+      options.push(currentlyEnteredDomain);
```

Comment:
I'm a bit hesitant about the modification of options here. Currently it's not a problem as `state.domains` is filtered in `getUnusedDomains` and returns a new array, but this could easily change in the future. This could then result in a hard to find state modification error.
Maybe change options to a `let` and then do `options = [...options, currentlyEnteredDomain];`

Reply:
Thanks for paying attention to this :+1: